packages/svelte/src/compiler/phases/3-transform/client/utils.js

```diff
@@ -184,6 +184,7 @@ export function get_prop_source(binding, state, name, initial) {
 
 	if (
 		state.analysis.accessors ||
+		binding.updated ||
```

Comment:
this makes the next three lines irrelevant, and has a false positive in the case where `immutable` is explicitly true

```suggestion
```